platform-sdk/swirlds-common/src/main/java/com/swirlds/common/concurrent/ExecutorFactory.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.concurrent;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * A factory for creating executors and threads.
+ */
+public interface ExecutorFactory {
```

Comment:
Can we put this into the platform context as a part of this PR? Outside the context it's not very easy to use. Put one of these in the context and the entire platform will exclusively use this utility for creating threads/executors within 48 hours. 😄